odoo/models.py

```diff
@@ -1535,20 +1535,27 @@ def default_get(self, fields_list):
                 defaults[name] = self._context[key]
                 continue
 
-            # 2. look up ir.default
-            if name in ir_defaults:
-                defaults[name] = ir_defaults[name]
+            field = self._fields.get(name)
+            if not field:
                 continue
 
-            field = self._fields.get(name)
+            # 2. look up default for non-company_dependent fields
+            if not field.company_dependent and name in ir_defaults:
+                defaults[name] = ir_defaults[name]
+                continue
 
             # 3. look up field.default
-            if field and field.default:
+            if field.default:
                 defaults[name] = field.default(self)
                 continue
 
-            # 4. delegate to parent model
-            if field and field.inherited:
+            # 4. look up fallback for company_dependent fields
```

Comment:
I think you should merge this into 2. So if you have a fallback or a default in ir_defaults it is used before the default property on self.

Reply:
It behaved like this before
the `field.default` is still used in business code, and I don't remove all of them especially for `default=lambda ...` for company_dependent fields

when field.default is defined and fallback is in the database

before: 
the fallback is defined in the `ir_property` which is not in the `defaults`
the value after create will be the result of `field.default`

now:
the fallback is also defined in the `ir_default` which is in the `defaults`
here if we don't do this, the value will be the fallback

Reply:
I saw that you have removed `default=...` from company dependent fields in business code because of wrong usage. Therefore maybe you could add a warning if default is used on a company dependent field? Then you could merge remove the `field.default` for company dependent fields (merge these 2).

I have not found any remaining usage in community and if there are some left in company, maybe they are invalid as well.

Reply:
I remvoed static `field.default` mainly because it is a bad practice. When the `field.default` and the `fallback` are different. (E.g. someone only changes the fallback by changing the ir.default reocrd), it will make most column value NOT NULL which is not good.

the remaining use cases are
`hr.payroll.structure.journal_id`
`product.template.responsible_id`